### PR TITLE
Add provision to setup Flask logging with filters for SentryHandler

### PR DESCRIPTION
**Pull request by arnavk**
Hi there! Would this be something you're willing to accept?

Add a provision to provide filters for SentryHandler while setting up integration with a Flask app. This is currently possible by poking around the SentryHandler after it has been configured in the `init_app` step but this patch could (maybe) make things a little convenient. If yes, then I can add tests and docs as well.

Usage:
```python
my_filter = SomeLoggingFilter()
sentry.init_app(app, logging_filters=my_filter)
```

### CODE REVIEW

**Comment by arnavk** [comment]
any opinions?

**Comment by dcramer** [comment]
I think it'd be better if we sent a signal, or something similar, so that you could do whatever you wanted here vs a one-off for filters.

File: raven/contrib/flask.py [hunk 1]
```diff
@@ -270,7 +277,11 @@ def init_app(self, app, dsn=None, logging=None, level=None,
             if self.logging_exclusions is not None:
                 kwargs['exclude'] = self.logging_exclusions
 
-            setup_logging(SentryHandler(self.client, level=self.level), **kwargs)
+            handler = SentryHandler(self.client, level=self.level)
+            setup_logging(handler, **kwargs)
+            if logging_filters is not None:
+                for _filter in logging_filters:
```
Comment (mitsuhiko) [inline raven/contrib/flask.py]: why is this called `_filter` and not `filter`?

Comment (arnavk) [inline raven/contrib/flask.py]: Oops, should have been `filter`

**Comment by arnavk** [comment]
@dcramer did you mean send a signal that the SentryHandler was added (with a reference to the handler?)

**Comment by dcramer** [comment]
@arnavk it could just be a signal that setup_logging was called with the details. Then you can do whatever you want. Its a lot more flexible and still very easy to instrument.

File: raven/contrib/flask.py (hunk 1 above)
Comment (arnavk) [inline raven/contrib/flask.py]: I remember now...`filter` is a function... so I just avoided that. Are you ok with `filter` as a name here or should I use `f` or something?

**Comment by arnavk** [comment]
@dcramer If I understand you correctly, once this signal is written, this PR should be replaced by a signal handler like so:

```
def init_app_called_handler(self, app, logging=None, **kwargs):
    if logging is not None:
        self.logging = logging

    if logging_exclusions is not None:
        self.logging_exclusions = logging_exclusions

    if logging_filters is not None:
        if not isinstance(logging_filters, list):
            logging_filters = [logging_filters]

        self.logging_filters = logging_filters
    if self.logging:
        kwargs = {}
        if self.logging_exclusions is not None:
            kwargs['exclude'] = self.logging_exclusions

        handler = SentryHandler(self.client, level=self.level)
        setup_logging(handler, **kwargs)
        if logging_filters is not None:
            for _filter in logging_filters:
                handler.addFilter(_filter)

```

Is that accurate?

**Comment by dcramer** [comment]
@arnavk I'm suggesting firing a signal so anyone can deal with it in whatever way they want, since this filters thing is fairly specific and we get to a point where configuration is too complex to solve ever need.

There's a couple of approaches we could do, though some have complexity:

1. We could fire a signal when Raven is configured
2. We could fire a signal when setup_logging is called
3. Flask-only: We could fire a signal in one or both of the above

The problem with 1/2 is that we'd need our own signal framework in place. It's not a complex thing, but it's probably overkill at this point. Whereas with #3, we can just require blinker like we already do, and wire, at the very least, setup_logging to fire something like a logging_configured signal. Your app could then just bind to that and add any filters/whatever from there.

**Comment by arnavk** [comment]
Ok, I'll rework this with #3. Since it'll likely be a bigger change, I'll make a separate PR and I'm closing this one for now. Thanks, @dcramer!

**Pull request closed without merging**